### DISCUSSION TITLE
VDFT probs do not sum up to one

### DISCUSSION THREAD

**Discussion by vinodkraman** [online-ml/river#915] [General]
I am trying to run VDFT on the balance-scale dataset. However, when I print the model at the end, the probabilities do not sum to 1. Also, I am noticing that under mc leaf prediction the accuracy is significantly lower compared to non mc leaf prediction. However, I'm not sure why this is the case. I've attached the code below. More specifically, its seems like the tree never splits...
[code.zip](https://github.com/online-ml/river/files/8508181/code.zip)

**Comment by smastelini** [comment]
Hi @vinodkraman, I did not observe the same problem. 

<img width="134" alt="image" src="https://user-images.githubusercontent.com/9000157/163976896-68b1dda6-873b-468b-8a39-5016422db3a6.png">

The majority class is indeed expected to perform worse than (adaptive) Naive Bayes models. This is especially true if no splits are performed.  The tree does not split because either:

- You have too few points
- There is not enough statistical evidence to enable splits

I suggest you to give a look in this [book chapter](https://book.moa.cms.waikato.ac.nz/chapter_2.html#sec2-11) for a starting point on decision trees. Also, you can refer to the [guide on Hoeffding Trees](https://riverml.xyz/latest/user-guide/on-hoeffding-trees/) that is available in the River website.

**Reply by vinodkraman** [reply]
hmm, there are only 625 examples in the dataset, how did your model get 825?

**Reply by smastelini** [reply]
I just ran your code using your data. I guess the model is using bootstrap sampling, as you are using HoeffdingAdaptiveTreeClassifier. It has a seed parameter to control the randomness of bootstrap sampling, when it is active (`bootstrap_sampling=True`).

**Reply by vinodkraman** [reply]
Oh shoot, totally missed this. Thanks for all the help!

**Reply by smastelini** [reply]
No worries!
